addons/udes_stock/models/stock_move_line.py

```diff
@@ -1110,14 +1115,35 @@ def _drop_off_criterion_summary(self):
         "qty_done",
     )
     def _prevent_write_after_done(self):
-        done_lines = self.filtered(lambda x: x.state == "done")
+        done_lines = self.filtered(lambda ml: ml.state == "done")
+
+        # When a moveline is updated through desktop only one "done" move line is passed into self even when there are multiple
+        # Get all the done move lines on the picking if there is only one "done" move line
+        if len(done_lines) == 1 and self.mapped("picking_id"):
+            done_lines = self.mapped("picking_id.move_line_ids").filtered(
+                lambda ml: ml.state == "done"
+            )
+
         if done_lines:
-            raise ValidationError(_("Cannot update move lines that are already 'done'."))
+            # For each done line
+            # Create error message
+            move_line_strings = list()
+            for line in done_lines:
+                _logger.error(
```

Comment:
Picking name and/or ID would be useful here too.

Reply:
changed